src/box/sql/mem.c

```diff
@@ -1748,88 +1748,38 @@ mem_cast_implicit_number(struct Mem *mem, enum field_type type)
 	return 0;
 }
 
-int
-mem_get_int(const struct Mem *mem, int64_t *i, bool *is_neg)
-{
-	if (mem->type == MEM_TYPE_INT) {
-		*i = mem->u.i;
-		*is_neg = true;
-		return 0;
-	}
-	if (mem->type == MEM_TYPE_UINT) {
-		*i = mem->u.i;
-		*is_neg = false;
-		return 0;
-	}
-	if ((mem->type & (MEM_TYPE_STR | MEM_TYPE_BIN)) != 0)
-		return sql_atoi64(mem->z, i, is_neg, mem->n);
-	if (mem->type == MEM_TYPE_DOUBLE) {
-		double d = mem->u.r;
-		if (d <= -1.0 && d >= (double)INT64_MIN) {
-			*i = (int64_t)d;
-			*is_neg = true;
-			return 0;
-		}
-		if (d > -1.0 && d < (double)UINT64_MAX) {
-			*i = (int64_t)(uint64_t)d;
-			*is_neg = false;
-			return 0;
-		}
-		return -1;
-	}
-	if (mem->type == MEM_TYPE_DEC) {
-		if (decimal_is_neg(&mem->u.d)) {
-			if (decimal_to_int64(&mem->u.d, i) == NULL)
-				return -1;
-			*is_neg = *i < 0;
-			return 0;
-		}
-		if (decimal_to_uint64(&mem->u.d, (uint64_t *)i) == NULL)
-			return -1;
-		*is_neg = false;
-		return 0;
-	}
-	return -1;
-}
-
-int
-mem_get_uint(const struct Mem *mem, uint64_t *u)
+int64_t
+mem_get_int_unsafe(const struct Mem *mem)
```

Comment:
What do you this about moving this patch in separate PR and dropping `mem_get_bool()`, `mem_get_bool_unsafe()`, `mem_get_bin()`, `mem_len()` and `mem_len_unsafe()`?

In this case it make sense to drop `mem_get_double_unsafe()` and just use `mem_get_double()` in the single place the `mem_get_double_unsafe()` is used.